taskpools/chase_lev_deques.nim

```diff
@@ -89,7 +87,7 @@ proc newBuf(T: typedesc, capacity: int): ptr Buf[T] =
     zero = true
   )
 
-  # result.prev = nil
+  # result.prevRetired = nil
```

Comment:
this is not needed because it's never read until garbage collection?

Reply:
I think I added that for completeness because all the other fields are initialized.
But previously there was no garbage collection so we didn't care if it was not initialized., hence commented out.

However now it definitely need to be initialized because `garbageCollect` will assume non-nil values need to be freed

Reply:
```suggestion
  result.prevRetired = nil
```